tests/dem/full_contact_functions.h

```diff
@@ -0,0 +1,327 @@
+                  Tensor<1, 3>                  &velocity1,
+                  Tensor<1, 3>                  &omega0,
+                  Tensor<1, 3>                  &omega1)
+{
+  velocity0[0] = particle0_properties[PropertiesIndex::v_x];
+  velocity0[1] = particle0_properties[PropertiesIndex::v_y];
+  velocity0[2] = particle0_properties[PropertiesIndex::v_z];
+  velocity1[0] = particle1_properties[PropertiesIndex::v_x];
+  velocity1[1] = particle1_properties[PropertiesIndex::v_y];
+  velocity1[2] = particle1_properties[PropertiesIndex::v_z];
+  omega0[0]    = particle0_properties[PropertiesIndex::omega_x];
+  omega0[1]    = particle0_properties[PropertiesIndex::omega_y];
+  omega0[2]    = particle0_properties[PropertiesIndex::omega_z];
+  omega1[0]    = particle1_properties[PropertiesIndex::omega_x];
+  omega1[1]    = particle1_properties[PropertiesIndex::omega_y];
+  omega1[2]    = particle1_properties[PropertiesIndex::omega_z];
+}
+
+using namespace Parameters::Lagrangian;
+
+/**
+ * @brief Simulate the full contact between two particles with chosen force model.
+ *
+ * @tparam dim Integer that denotes the number of spatial dimensions.
+ * @tparam PropertiesIndex Index of the properties used within the ParticleHandler.
+ * @tparam contact_model Model for the contact force.
+ * @tparam rolling_friction_model Rolling resistance model.
+ * @param triangulation Triangulation to access the information of the cells.
+ * @param particle_handler Storage of particles and their accessor functions.
+ * @param contact_manager Manages the contact between particles.
+ * @param dem_parameters DEM parameters declared.
+ * @param p Initial properties of particles.
+ * @param g A constant volumetric body force applied to all particles
+ * @param dt Time Step.
+ * @param output_step Step at which force and overlap are written on file.
+ * @param neighborhood_threshold Threshold value of contact detection.
+ * @param filename Name of file where force and overlap are written.
+ *
+ * @return Struct of vectors of time, force and torque acting on particle 0, overlap, tangential_overlap, velocities, angular velocities.
+ */
+template <int dim,
+          typename PropertiesIndex,
+          ParticleParticleContactForceModel contact_model,
+          RollingResistanceMethod           rolling_friction_model>
+full_contact_output
+simul_full_contact(parallel::distributed::Triangulation<dim> &triangulation,
+                   Particles::ParticleHandler<dim>           &particle_handler,
+                   DEMContactManager<dim, PropertiesIndex>   &contact_manager,
+                   DEMSolverParameters<dim>                  &dem_parameters,
+                   initial_particle_properties<dim>          &p,
+                   Tensor<1, 3>                              &g,
+                   double                                     dt,
+                   unsigned int                               output_step,
```

Comment:
Do you need the output_step unsigned integer? In the end you always output the full contact right?